### PR TITLE
Testing the `s` limits in `ECRECOVER`

### LINKED ISSUE #2064: Testing the `s` limits in `ECRECOVER`
We have to expand our test suite for `ECRECOVER` to test `s` against its limits. Recall that
```
s =
  | (secp256k1n - 1) / 2 - 1   // universally accepted
  | (secp256k1n - 1) / 2       // universally accepted
  | (secp256k1n - 1) / 2 + 1   // universally accepted
  | (secp256k1n - 1) / 2 + 2   // rejected for transactions, acceptable in ECRECOVER
  | 
  | secp256k1n - 2   // acceptable in ECRECOVER
  | secp256k1n - 1   // acceptable in ECRECOVER
  | 
  | secp256k1n       // universally rejected
  | secp256k1n + 1   // universally rejected

```

### PR DESCRIPTION
**Pull request by lorenzogentile404**
> [!NOTE]
> Adds ECRECOVER s-boundary test cases and makes successBit assertion conditional via nullable expectations.
> 
> - **Tests**:
>   - **ECRECOVER s-limit coverage**: Add `ecRecoverSourceForSLimits()` to generate edge cases around `SECP256K1N` and include it in `@MethodSource` for both standard and nightly parameterized tests.
>   - **Assertions**: Switch expected flags to `Boolean` and assert `successBit` only when non-null to accommodate s-limit cases where success is not required.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 0cca7a5af82cfb8082b2b5919946c29ab0e7bb81. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: arithmetization/src/test/java/net/consensys/linea/zktracer/module/ecdata/EcRecoverTest.java [hunk 1]
```diff
@@ -219,6 +222,41 @@ private static Stream<Arguments> ecRecoverSource() {
...
+            ((SECP256K1N.subtract(1)).divide(2)).subtract(1), // universally accepted
+            ((SECP256K1N.subtract(1)).divide(2)), // universally accepted
+            ((SECP256K1N.subtract(1)).divide(2)).add(1), // universally accepted
+            ((SECP256K1N.subtract(1)).divide(2))
+                .add(2), // rejected for transactions, acceptable in ECRECOVER
+            SECP256K1N.subtract(2), // acceptable in ECRECOVER
+            SECP256K1N.subtract(1), // acceptable in ECRECOVER
+            SECP256K1N, // universally rejected
+            SECP256K1N.add(1)); // universally rejected
+
```
Comment (lorenzogentile404) [inline arithmetization/src/test/java/net/consensys/linea/zktracer/module/ecdata/EcRecoverTest.java:292]: Add documentation.

**Review by OlivierBBB — APPROVED** [review]
LGTM
